### ISSUE TITLE
Commuter Testing should not be coupled with dynamic NPIs

### ISSUE THREAD

**Issue by HenrZu** [SciCompMod/memilio#1116]
### Motivation / Current Behaviour

in the advance function of the secirvvs model, we always set
```
this->get_model().parameters.get_start_commuter_detection() = (ScalarType)t_start;
this->get_model().parameters.get_end_commuter_detection()   = (ScalarType)t_end;
```

Testing should not necessarily be coupled to the dynamic NPIs in this way. Also, we can't change the non detection value during the simulation.

### Enhancement description

make the code more modular and split up features instead of combining them.

### Checklist

- [X] Attached labels, especially loc:: or model:: labels.
- [X] Linked to project

**Comment by mknaranja** [comment]
Depends on #1523

**Comment by HenrZu** [comment]
As of PR #1523 , commuter detection start/end is still set inside the dynamic NPI block in osecirvvs, osecirts, and osecir models. The coupling has not been resolved yet. This remains open for a follow-up PR.

**Comment by mknaranja** [comment]
True. I would however prefer to allow a selection (via a function/macro/definition or whatever in the code base) if commuter testing is or is not coupled with DynamicNPIs as this is also some type of dynamic intervention.
